src/vs/workbench/contrib/positronVariables/browser/components/variablesInstanceMenuButton.tsx

```diff
@@ -27,19 +28,36 @@ export const VariablesInstanceMenuButton = () => {
 	// Helper method to calculate the label for a runtime.
 	const labelForRuntime = (session?: ILanguageRuntimeSession): string => {
 		if (session) {
-			return session.metadata.sessionName;
+			return session.getLabel();
 		}
 		return 'None';
 	};
 
+	// Store just the label in state instead of the entire session
+	const [sessionLabel, setSessionLabel] = useState<string>(
+		labelForRuntime(positronVariablesContext.activePositronVariablesInstance?.session)
+	);
+
+	// Use an effect to update the session label when active instance changes
+	useEffect(() => {
+		const disposables = new DisposableStore();
+
+		// Update label when active instance changes
+		disposables.add(positronVariablesContext.positronVariablesService.onDidChangeActivePositronVariablesInstance(instance => {
+			setSessionLabel(labelForRuntime(instance?.session));
+		}));
+
+		return () => disposables.dispose();
+	}, [positronVariablesContext.positronVariablesService, positronVariablesContext.runtimeSessionService, positronVariablesContext.activePositronVariablesInstance]);
+
 	// Builds the actions.
 	const actions = () => {
 		// Build the actions.
 		const actions: IAction[] = [];
 		positronVariablesContext.positronVariablesInstances.map(positronVariablesInstance => {
 			actions.push({
 				id: positronVariablesInstance.session.sessionId,
-				label: positronVariablesInstance.session.metadata.sessionName,
+				label: labelForRuntime(positronVariablesInstance.session),
```

Comment:
Should this also use `sessionLabel`?
```suggestion
				label: sessionLabel,
```

Reply:
Good catch!